etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,549 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PacemakerCluster is used in Two Node OpenShift with Fencing (TNF) deployments to monitor the health
+// of etcd running under pacemaker.
+
+// PacemakerCluster condition types (cluster-level)
+const (
+	// PacemakerClusterHealthyConditionType tracks the overall health of the pacemaker cluster.
+	// This is an aggregate condition that reflects the health of all cluster-level conditions and node health.
+	// Specifically, it aggregates the following conditions:
+	// - PacemakerClusterNotInMaintenanceModeConditionType
+	// - PacemakerClusterExpectedNodeCountConditionType
+	// - PacemakerClusterNodeHealthyConditionType (for each node)
+	// When True, the cluster is healthy with reason "ClusterHealthy".
+	// When False, the cluster is unhealthy with reason "ClusterUnhealthy".
+	PacemakerClusterHealthyConditionType = "PacemakerClusterHealthy"
+
+	// PacemakerClusterNotInMaintenanceModeConditionType tracks whether the cluster is not in maintenance mode.
+	// Maintenance mode is a cluster-wide setting that prevents pacemaker from starting or stopping resources.
+	// When True, the cluster is not in maintenance mode with reason "NotInMaintenance". This is the normal operating state.
+	// When False, the cluster is in maintenance mode with reason "InMaintenance". This is an unexpected state.
+	PacemakerClusterNotInMaintenanceModeConditionType = "PacemakerClusterNotInMaintenanceMode"
```

Comment:
I would consider negating this so that the expected state is false and you don't have negation in the name